### DISCUSSION TITLE
icanhazstring/composer-unused (installed globally) contains a Composer plug in which is blocked by your allow-plugins config

### DISCUSSION THREAD

**Discussion by ilhanyumer** [composer-unused/composer-unused#377] [Q&A] [answered]
When I try to run Composer > Install from PHP Storm I get the following error:

    In PluginManager.php line 762:
                                                                    
    icanhazstring/composer-unused (installed globally) contains a Composer plug  
    in which is blocked by your allow-plugins config. You may add it to the lis  
    t if you consider it safe.                                                   
    You can run "composer global config --no-plugins allow-plugins.icanhazstrin  
    g/composer-unused [true|false]" to enable it (true) or disable it explicitl  
    y and suppress this exception (false)                                        
    See https://getcomposer.org/allow-plugins    

What might be the reason for this error message above?

I have tried to run

    /etc/php8/bin/php composer.phar global config --no-plugins allow-plugins.icanhazstrin  true

and I have also tried to add the following to the `composer.json` file

        "config": {
            "allow-plugins": true,

but still I get the same error message.

My `composer.json` is something like:

    {
        "name": "test/version1",
        "description":"Test Version 1",
        "keywords":[],
        "config": {
            "allow-plugins": true,
            "secure-http":false
        },
        "scripts": {
            "test": "phpunit --config tests/phpunit.xml"
        },
	    "require": {
	 	    "php": ">=7.3",
            "monolog/monolog": "1.26.*",

    ...

        "require-dev": {
            "php": ">=8.1",
            "phpunit/phpunit": "^9.5.0",
            "friendsofphp/php-cs-fixer": "^2.16",
            "icanhazstring/composer-unused": "^0.7.5"
        }
    }

**Comment by icanhazstring** [comment]
Hi thanks for the report.
AFAIK the config should look like this
https://getcomposer.org/doc/06-config.md#allow-plugins

So I don't know why doesn't work. Seems not really composer-unused related.

Also latest version doesn't even use plugin integration anymore. This is provided using composer-unused/composer-unused-plugin

**Comment by ilhanyumer** [comment, marked as answer]
{
        "config": {
            "allow-plugins": false
        }
    }

The code above didn't solve my problem after I wrote it into my `composer.json` file. Apparently there is another `composer.json` file located at `/home/yourUbuntuUsername/.config/composer` that needs to be updated. I have solved my problem by running the command below:

    /etc/php8/bin/php composer.phar global show -i

Then it asked me if I trust `icanhazstring/composer-unused` and I said `y`. Then my problem was solved.

**Reply by icanhazstring** [reply]
Glad you found a solution 👍
